source/docs/contracts_configuration.md

```diff
@@ -1,322 +1,327 @@
+<pre><code class="javascript">...
+contracts: {
+  SimpleStorage: {
+    args: [100, '$OtherContract']
+  },
+  OtherContract: {...}
+}
+...
+</code></pre>
+
+## Disabling deployment
+
+Sometimes we want to configure different behaviour for certain contracts within different [environments](environments.html). One of those cases could be that we don't actually want to deploy `SimpleStorage` in the production environment as we might expect some other storage Smart Contract to already be somewhere out there.
+
+We can prevent Embark from deploying any of our Smart Contracts by using the `deploy` configuration and setting it to `false` like this:
+
+<pre><code class="javascript">...
+development: 
+  contracts: {
+    SimpleStorage: {
+      args: [100]
+    }
+  }
+},
+production: {
+  contracts: {
+    SimpleStorage: {
+      deploy: false
     }
   }
-  // ...
 }
+...
 </code></pre>
 
 
-### Specify contract file
+## Deploying multiple instances
 
-By default embark will look for the contracts inside the folder defined in the `"contracts"` property in `embark.json`.
-However, it's possible to specify the contract filepath:
+In cases where we want to create multiple instances of the same Smart Contract but with, for example, different initialization values per instance, we can use the `instanceOf` property and refer to the original Smart Contract that should be deployed multiple times.
 
-<pre><code class="javascript">// config/contracts.js
-module.exports = {
-  "development": {
-    "gas": "auto",
-    "contracts": {
-      "SimpleStorage": {
-        <mark class="highlight-inline">"file": "./some_folder/simple_storage.sol",</mark>
-        "args": [
-          100
-        ]
-      }
-    }
+This can then be combined with [disabling the deployment](#Disabling-deployment) of the original Smart Contract like this:
+
+<pre><code class="javascript">...
+contracts: {
+  Currency: {
+    deploy: false,
+    args: [100]
+  },
+  Usd: {
+    instanceOf: 'Currency',
+    args: [200]
+  },
+  MyCoin: {
+    instanceOf: 'Currency',
+    args: [300]
+  }
+}
+...</code></pre>
+
+In the example above, we deploy `Usd` and `MyCoin` as instances of `Currency`. Notice that `Currency` itself isn't going to be deployed but merely functions as a "recipe" to create other instances of it.
+
+## Referencing already deployed Smart Contracts
+
+Embark not only integrates with the Smart Contracts that we create and own, but also with Smart Contracts that are already deployed and potentially owned by third-parties. If we wish to reference such a Smart Contract instance, all we have to do is specify the `address` of the instance in question.
+
+The following example configures `UserStorage` to be a Smart Contract instance that's already deployed and a dependency of our own `UserManagement` Smart Contract at the same time.
+
+<pre><code class="javascript">...
+contracts: {
+  UserStorage: {
+    address: '0x123456'
+  },
+  UserManagement: {
+    args: ['$UserStorage']
   }
```

Comment:
The UserManagement part is useless IMO. We already stressed enough the dependency on contracts.